rmw_opensplice_cpp/src/rmw_service_names_and_types.cpp

```diff
@@ -119,7 +120,13 @@ rmw_get_service_names_and_types(
       // Duplicate and store each type for the service
       size_t type_index = 0;
       for (const auto & type : service_n_types.second) {
-        char * type_name = rcutils_strdup(type.c_str(), *allocator);
+        size_t n = type.find(SAMPLE_PREFIX);
+        std::string stripped_type = type;
+        if (n < type.size() - strlen(SAMPLE_PREFIX)) {
+          stripped_type = std::string(type.substr(0, n + 1).c_str()) + \
```

Comment:
`type.substr()` already returns a `std::string` so no need to std::string(result.c_str())` the result.

Same on the next line.